src/modules/forces/functions.cpp

```diff
@@ -234,9 +182,57 @@ void ForcesModule::totalForces(ProcessPool &procPool, Configuration *cfg, const
 void ForcesModule::totalForces(ProcessPool &procPool, Species *sp, const PotentialMap &potentialMap,
                                std::vector<Vec3<double>> &f)
 {
+    // Zero force array
     std::fill(f.begin(), f.end(), Vec3<double>());
 
-    interAtomicForces(procPool, sp, potentialMap, f);
+    double scale, r, magjisq;
+    const auto cutoffSq = potentialMap.range() * potentialMap.range();
+    Vec3<double> vecij;
+    // NOTE PR #334 : use for_each_pair
+    for (auto indexI = 0; indexI < sp->nAtoms() - 1; ++indexI)
+    {
+        auto &i = sp->atom(indexI);
+
+        for (auto indexJ = indexI + 1; indexJ < sp->nAtoms(); ++indexJ)
+        {
+            auto &j = sp->atom(indexJ);
+
+            // Get intramolecular scaling of atom pair
+            scale = i.scaling(&j);
+            if (scale < 1.0e-3)
+                continue;
+
+            // Determine final forces
+            vecij = j.r() - i.r();
+            magjisq = vecij.magnitudeSq();
+            if (magjisq > cutoffSq)
+                continue;
+            r = sqrt(magjisq);
+            vecij /= r;
+
+            vecij *= potentialMap.force(&i, &j, r) * scale;
+            f[indexI] += vecij;
+            f[indexJ] -= vecij;
+        }
+    }
```

Comment:
```suggestion
    dissolve::for_each_pair(ParallelPolicies::seq, sp->atoms.begin(), sp->atoms.end(), [&](int indexI, auto &i, int indexJ, auto&j) {
        if (i == j) return;
        
        // Get intramolecular scaling of atom pair
        scale = i.scaling(&j);
        if (scale < 1.0e-3) return;

        // Determine final forces
        vecij = j.r() - i.r();
        magjisq = vecij.magnitudeSq();
        if (magjisq > cutoffSq) return;
        r = sqrt(magjisq);
        vecij /= r;

        vecij *= potentialMap.force(&i, &j, r) * scale;
        f[indexI] += vecij;
        f[indexJ] -= vecij;
    });
```

I don't think that this is safe to run in parallel, since we might get a race condition on different threads updating the forces.

Reply:
You're probably right - I'll implement a combinable here.